src/utils.jl

```diff
@@ -255,3 +255,6 @@ prob2dtmin(tspan, onet, ::Any) = onet*1//10^(10)
 timedepentdtmin(integrator::DEIntegrator) = timedepentdtmin(integrator.t, integrator.opts.dtmin)
 timedepentdtmin(t::AbstractFloat, dtmin) = abs(max(eps(t), dtmin))
 timedepentdtmin(::Any, dtmin) = abs(dtmin)
+
+maybe_with_logger(f, logger) = logger === nothing ? f() : with_logger(f, logger)
+default_logger() = Juno.isactive() ? nothing : TerminalLogger()
```

Comment:
Maybe

```suggestion
default_logger() = Juno.isactive() || current_logger() isa TerminalLogger ? nothing : TerminalLogger()
```

or, even more hacky "solution" that does not require `Juno`:

```suggestion
default_logger() =
    Logging.min_enabled_level(current_logger()) == LogLevel(-1) ? nothing : TerminalLogger()
```

Reply:
how does the latter work with Juno?

Reply:
Actually, I think @devmotion's solution using `LoggingExtras.TeeLogger` is much better https://github.com/JuliaDiffEq/DifferentialEquations.jl/issues/424#issuecomment-596263880 though I don't think you need to use a macro.

Reply:
> how does the latter work with Juno?

The log level of the `JunoProgressLogger` is guaranteed to be <= -1: https://github.com/JunoLab/Atom.jl/blob/016327893e4c2a22688446758debd1a4d47ed721/src/progress.jl#L75-L76